src/com/amazon/ion/impl/IonCursorBinary.java

```diff
@@ -178,8 +262,362 @@ class IonCursorBinary implements IonCursor {
         this.offset = offset;
         this.limit = offset + length;
         byteBuffer = ByteBuffer.wrap(buffer, offset, length);
+        refillableState = null;
+    }
+
+    /**
+     * The standard {@link IonBufferConfiguration}. This will be used unless the user chooses custom settings.
+     */
+    private static final IonBufferConfiguration STANDARD_BUFFER_CONFIGURATION =
+        IonBufferConfiguration.Builder.standard().build();
+
+    /**
+     * @param value a non-negative number.
+     * @return the exponent of the next power of two greater than or equal to the given number.
+     */
+    private static int logBase2(int value) {
+        return 32 - Integer.numberOfLeadingZeros(value == 0 ? 0 : value - 1);
+    }
+
+    /**
+     * @param value a non-negative number
+     * @return the next power of two greater than or equal to the given number.
+     */
+    static int nextPowerOfTwo(int value) {
+        return (int) Math.pow(2, logBase2(value));
+    }
+
+    /**
+     * Cache of configurations for fixed-sized streams. FIXED_SIZE_CONFIGURATIONS[i] returns a configuration with
+     * buffer size max(8, 2^i). Retrieve a configuration large enough for a given size using
+     * FIXED_SIZE_CONFIGURATIONS(logBase2(size)). Only supports sizes less than or equal to
+     * STANDARD_BUFFER_CONFIGURATION.getInitialBufferSize().
```

Comment:
```suggestion
     * STANDARD_BUFFER_CONFIGURATION.getMaximumBufferSize().
```

or do I misunderstand?

Reply:
It's correct as-written. This limits us to `logBase2(STANDARD_BUFFER_CONFIGURATION)` fixed-size configurations (i.e., 15). We have to draw the line somewhere. This keeps the footprint of FIXED_SIZE_CONFIGURATIONS small and prevents us from using fixed-size configurations that require allocating really large initial buffers. Say the user provides a ByteArrayInputStream backed by a 2GB `byte[]` -- in that case, even though the data is fixed, we don't want to allocate another 2 GB buffer to copy into. Unless the user provides a custom buffer configuration, we'll just use the standard one that starts at 32K and copy into it incrementally as needed.

Reply:
Added.